### PR TITLE
implement endpoint for starting accessioning

### PR DESCRIPTION
**Pull request by peetucket**
## Why was this change made?

part of #702 to allow for reaccessioning via Goobi
when done, return to the checklist in #702 

## Was the API documentation (openapi.yml) updated?

yes

## Does this change affect how this application integrates with other services?

If so, please confirm:
- [ ] change was tested on stage    and/or
- [ ] test added to sul-dlss/infrastructure-integration-test

### CODE REVIEW

File: app/controllers/objects_controller.rb [hunk 1]
```diff
@@ -50,6 +50,28 @@ def show
     render json: Cocina::Mapper.build(@item)
   end
 
+  # Initialize specified workflow (assemblyWF by default), and also version if needed
+  # called by pre-assembly, goobi and lybservices-scripts to kick off accessioning for a new or existing object
+  # params can optionally include the information needed to open a new version as well as the workflow to start (which defaults to assemblyWF)
+  # @option opts [Boolean] :assume_accessioned If true, does not check whether object has been accessioned.
```
Comment (jcoyne) [inline app/controllers/objects_controller.rb]: This method doesn't have any options.

File: app/controllers/objects_controller.rb [hunk 2]
```diff
@@ -50,6 +50,28 @@ def show
...
+  # @option opts [String] :significance set significance (major/minor/patch) of version change
+  # @option opts [String] :description set description of version change
+  # @option opts [String] :opening_user_name add opening username to the events datastream
+  # @option opts [String] :workflow the workflow to start (defaults to 'assemblyWF')
+  def start_accession
+    params[:workflow] ||= default_start_accession_workflow # default if not specified
```
Comment (jcoyne) [inline app/controllers/objects_controller.rb]: can this be `workflow = params[:workflow] || default_start_accession_workflow`?

File: config/routes.rb [hunk 3]
```diff
@@ -27,6 +27,7 @@
         post 'preserve'
         post 'update_marc_record'
         post 'notify_goobi'
+        post 'start_accession'
```
Comment (jcoyne) [inline config/routes.rb]: Not a big deal, but how about just `accession`?

File: openapi.yml [hunk 4]
```diff
@@ -175,6 +175,32 @@ paths:
...
+    post:
+      tags:
+        - objects
+      summary: Start accessioning or re-accession an object to DOR
+      description: ''
+      operationId: 'objects#start_accession'
+      responses:
+        '201':
+          description: Accessiong/re-accessioning started
+      parameters:
```
Comment (jcoyne) [inline openapi.yml]: Are significance, description, opening_user_name missing?

File: openapi.yml [hunk 5]
```diff
@@ -175,6 +175,32 @@ paths:
...
+        - name: id
+          in: path
+          description: ID of object
+          required: true
+          schema:
+            $ref: '#/components/schemas/Druid'
+        - name: workflow
+          in: query
+          description: which workflow should be started
```
Comment (jcoyne) [inline openapi.yml]: add a note about the default.

File: app/controllers/objects_controller.rb (hunk 1 above)
Comment (ndushay) [reply app/controllers/objects_controller.rb]: yeah, maybe reword indicated they are params passed to the controller?

File: app/controllers/objects_controller.rb [hunk 6]
```diff
@@ -94,6 +120,10 @@ def create_params
     params.except(:action, :controller).to_unsafe_h.merge(body_params)
   end
 
+  def default_start_accession_workflow
```
Comment (ndushay) [inline app/controllers/objects_controller.rb]: I'm 'meh' on calling this out as a method;  as a former java-head, i would define it as a string near the method that uses it, or just use it directly in the `||=` statement.  But leaving it as is works too.

File: spec/requests/start_accession_spec.rb [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+      allow(VersionService).to receive(:can_open?).and_return(false)
+    end
+
+    it 'does not open or close a version and starts default workflow' do
+      post "/v1/objects/#{druid}/start_accession",
+           headers: { 'Authorization' => "Bearer #{jwt}" }
+      expect(response).to be_successful
+      expect(workflow_client).to have_received(:create_workflow_by_name).with(object.pid, default_start_accession_workflow, version: '1')
+      expect(VersionService).not_to have_received(:open)
+      expect(VersionService).not_to have_received(:close)
```
Comment (ndushay) [inline spec/requests/start_accession_spec.rb]: i would probably split the "what it does re: version" into a separate test and ignore it in this test, but meh - no biggie.

File: app/controllers/objects_controller.rb (hunk 1 above)
Comment (peetucket) [reply app/controllers/objects_controller.rb]: will update comment to indicate those params are passed through to versioning (they are useful so an app like pre-assembly can specify then).

File: app/controllers/objects_controller.rb (hunk 2 above)
Comment (peetucket) [reply app/controllers/objects_controller.rb]: sounds good

File: config/routes.rb (hunk 3 above)
Comment (peetucket) [reply config/routes.rb]: sure

File: openapi.yml (hunk 5 above)
Comment (peetucket) [reply openapi.yml]: done

File: openapi.yml (hunk 4 above)
Comment (peetucket) [reply openapi.yml]: added
